src/js/config/Constants.js

```diff
@@ -1,5 +1,15 @@
-import { Platform } from 'react-native';
+import { Platform, Dimensions } from 'react-native';
 
-export const IS_IOS = Platform.OS === 'ios';
 export const TEST_PREFIX = 'test-';
 export const MAKE_ACCESSIBLE_FOR_AUTOMATION = { accessible: false };
+export const IS_IOS = Platform.OS === 'ios';
+export const {
+  width: WINDOW_WIDTH,
+  height: WINDOW_HEIGHT,
+} = Dimensions.get('window');
+export const IS_IPHONEX = IS_IOS &&
+  !Platform.isPad &&
+  !Platform.isTVOS &&
+  ((WINDOW_WIDTH === 812 || WINDOW_HEIGHT === 812) || (WINDOW_WIDTH === 896 || WINDOW_HEIGHT === 896));
```

Comment:
I have no idea how this statement works or the logic behind it - I'd have to dissect this based on precedence order of brackets combined with searching online to understand the details of how this determines whether this is an iPhone X. Can you please add a comment explaining the logic here for readers of the codebase?

Reply:
Added a comment for it